### PR TITLE
Clippy fixes

### PR DESCRIPTION
**Pull request by julienr**
Fixes a number of clippy issues:
- Implement `Default` for various types
- Remove some useless format/mut
- Simplify some `if let match`
- Fix some cases where variables are never read because they are immediatly overwritten

### CODE REVIEW

**Review by nyurik — APPROVED** [review]
lgtm

**Comment by julienr** [comment]
Looks like the build is failing because some dependencies require rust 1.65:
```
Caused by:
  package `wgpu-hal v0.14.0 (/home/runner/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/d536613/wgpu-hal)` cannot be built because it requires rustc 1.65 or newer, while the currently active rustc version is 1.64.0
```
I guessed this is blocked by #201 then ?

**Comment by nyurik** [comment]
@julienr :shrug: I think @maxammann would have to answer that. Or you could try to tackle #201?

**Comment by maxammann** [comment]
Its a new issue :) Ill fix it today propably.

**Comment by maxammann** [comment]
I'll merge as soon as checks succeed: https://github.com/maplibre/maplibre-rs/pull/207

File: maplibre/src/io/geometry_index.rs [hunk 1]
```diff
@@ -62,6 +62,12 @@ impl GeometryIndex {
     }
 }
 
+impl Default for GeometryIndex {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```
Comment (maxammann) [inline maplibre/src/io/geometry_index.rs:69]: I was wondering which was is better: 
* Implement the constructor in new()
* or implement it in default() and then call default() in new()

What do you think?

**Review by maxammann — APPROVED** [review]

File: maplibre/src/io/geometry_index.rs (hunk 1 above)
Comment (julienr) [reply maplibre/src/io/geometry_index.rs:69]: No strong feeling and I'm not a Rust expert. But in the [clippy docs](https://rust-lang.github.io/rust-clippy/master/#new_without_default) they seem to point ot this pattern (just have default delegate to new)

Comment (maxammann) [reply maplibre/src/io/geometry_index.rs:69]: Oke cool, that let's use that style!

**Pull request merged**